src/lib/FreeC/Frontend/Haskell/Simplifier.hs

```diff
@@ -480,11 +482,30 @@ simplifyTypeSchema (HSE.TyForall srcSpan (Just binds) Nothing typeExpr) = do
 -- Without explicit @forall@.
 simplifyTypeSchema typeExpr = do
   typeExpr' <- simplifyType typeExpr
-  let srcSpan  = IR.typeSrcSpan typeExpr'
-      typeArgs = map
-        (IR.TypeVarDecl NoSrcSpan . fromJust . IR.identFromQName)
-        (freeTypeVars typeExpr')
+  let srcSpan         = IR.typeSrcSpan typeExpr'
+      typeArgIdents   = freeTypeVars typeExpr'
+      typeArgSrcSpans = map (findTypeArgSrcSpan typeExpr') typeArgIdents
+      typeArgs        = zipWith IR.TypeVarDecl typeArgSrcSpans typeArgIdents
   return (IR.TypeSchema srcSpan typeArgs typeExpr')
+ where
+  -- | Finds the first occurrence of the type variable with the given name.
+  --
+  --   Returns 'NoSrcSpan' if 'findTypeArgSrcSpan'' returns @Nothing@.
+  --   Since the type arguments have been extracted using 'freeTypeVars',
+  --   'findTypeArgSrcSpan'' should never return @Nothing@.
+  findTypeArgSrcSpan :: IR.Type -> IR.TypeVarIdent -> SrcSpan
+  findTypeArgSrcSpan = fromMaybe NoSrcSpan .: flip findTypeArgSrcSpan'
+
+  -- | Like 'findTypeArgSrcSpan' but returns @Nothing@ if there is
```

Comment:
According to [this article](https://www.grammarly.com/blog/comma-before-but/) a comma is put before *but* only if it connects two independent clauses. 
If the comment was `This function behaves like 'findTypeArgSrcSpan', but it returns @Nothing@ if there is no such type variable.`, I'd agree that a comma should be placed before *but* since both halves could be sentences on their own. However, in this case I'm not sure what's the correct punctuation. `Like 'findTypeArgSrcSpan'` doesn't sound like a full sentence to me. However, I could be wrong. @sad what do you think is the proper punctuation in this case?

Reply:
I also looked at this article during my review and think that the comma should be there:
`Like 'findTypeArgSrcSpan'` may not be a full English sentence, but it this context it stands for a full sentence like the one you described and is therefore an independent clause.
This can also be seen by assuming `Like 'findTypeArgSrcSpan'` were the entire comment for this function. While it would be strange to have two functions doing the same thing from a programming point of view, this would still be a sensible comment from a linguistic point of view.